spark/src/main/scala/com/alibaba/graphar/writer/EdgeWriter.scala

```diff
@@ -31,75 +31,92 @@ import scala.collection.mutable.ArrayBuffer
 
 /** Helper object for EdgeWriter class. */
 object EdgeWriter {
-  // split the whole edge dataframe into chunk dataframes by vertex chunk size.
-  private def split(edgeDf: DataFrame, keyColumnName: String, vertexChunkSize: Long): Seq[DataFrame] = {
-    // split the dataframe to mutiple daraframes by vertex chunk
-    edgeDf.cache()
-    val spark = edgeDf.sparkSession
+  private def repartitionAndSort(spark: SparkSession,
+                                 edgeDf: DataFrame,
+                                 edgeInfo: EdgeInfo,
+                                 adjListType: AdjListType.Value,
+                                 vertexNumOfPrimaryVertexLabel: Long): (DataFrame, Seq[DataFrame], Array[Long]) = {
     import spark.implicits._
-    val df_schema = edgeDf.schema
-    val index = df_schema.fieldIndex(keyColumnName)
-    val vertex_chunk_num = math.floor(edgeDf.agg(max(keyColumnName)).head().getLong(0) / vertexChunkSize.toDouble).toInt
-    val chunks: Seq[DataFrame] = (0 to vertex_chunk_num).map {i => edgeDf.where(edgeDf(keyColumnName) >= (i * vertexChunkSize)  and edgeDf(keyColumnName) < ((i + 1) * vertexChunkSize))}
-    return chunks
-  }
-
-  // repartition the chunk dataframe by edge chunk size (this is for COO)
-  private def repartition(chunkDf: DataFrame, keyColumnName: String, edgeChunkSize: Long): DataFrame = {
-    // repartition the dataframe by edge chunk size
-    val spark = chunkDf.sparkSession
-    import spark.implicits._
-    val df_schema = chunkDf.schema
-    val index = df_schema.fieldIndex(keyColumnName)
-    val df_rdd = chunkDf.rdd.map(row => (row(index).asInstanceOf[Long], row))
-
-    // generate global edge id for each record of dataframe
-    val parition_counts = df_rdd
+    val edgeSchema = edgeDf.schema
+    val colIndex = edgeSchema.fieldIndex(if (adjListType == AdjListType.ordered_by_source) GeneralParams.srcIndexCol else GeneralParams.dstIndexCol)
```

Comment:
Is this function repartitionAndSort() only required for AdjListType.ordered_by_source & AdjListType.ordered_by_dest? It seems that it is called for all types.